web/src/site-admin/SiteAdminActivation.tsx

```diff
@@ -0,0 +1,259 @@
+                                title="Enable repositories"
+                                done={this.props.checklistInfo.enabledRepository}
+                                action={this.enableRepos}
+                                detail="Select which repositories Sourcegraph should pull and index from your code host(s)."
+                            />
+                        </li>
+                        <li>
+                            <ChecklistItem
+                                title="Enable an extension"
+                                done={this.props.checklistInfo.enabledExtension}
+                                action={this.enableExtension}
+                                detail="Enable a Sourcegraph extension to add jump-to-def, find-refs, or helpful annotations from 3rd party dev tools."
+                            />
+                        </li>
+                        <li>
+                            <ChecklistItem
+                                title="Configure sign-up"
+                                done={this.props.checklistInfo.enabledSignOn}
+                                action={this.enableSignOn}
+                                detail="Configure a single-sign on (SSO) provider or enable open sign-up so you can share Sourcegraph with others on your team."
+                            />
+                        </li>
+                    </ul>
+                ) : (
+                    <div>Loading...</div>
+                )}
+            </div>
+        )
+    }
+}
+
+interface ChecklistItemProps {
+    title: string
+    action: () => void
+    done?: boolean
+    detail?: string
+}
+
+interface ChecklistItemState {
+    showDetail?: boolean
+}
+
+/**
+ * ChecklistItem is a single item in the site admin activation checklist.
+ */
+class ChecklistItem extends React.PureComponent<ChecklistItemProps, ChecklistItemState> {
+    public state: ChecklistItemState = {}
+
+    private toggleDetail = (e: React.MouseEvent<HTMLSpanElement>) => {
+        e.stopPropagation()
+        this.setState({ showDetail: !this.state.showDetail })
+    }
+
+    public render(): JSX.Element {
+        return (
+            <div className="item-container">
+                <p className="item-title" onClick={this.props.action}>
+                    {this.props.done ? (
+                        <CheckboxMarkedCircleOutlineIcon className="icon-inline done" />
+                    ) : (
+                        <CheckboxBlankCircleOutlineIcon className="icon-inline todo" />
+                    )}
+                    &nbsp;&nbsp;
+                    {this.props.title}
+                    &nbsp;
+                    <span className="info-button" onClick={this.toggleDetail}>
+                        <InformationOutlineIcon className="icon-inline" />
+                    </span>
+                </p>
+                <p className={'item-detail' + (this.state.showDetail ? ' show' : ' hide')}>{this.props.detail}</p>
```

Comment:
Not necessary, but you could also use a template string
```suggestion
                <p className={`item-detail ${this.state.showDetail ? 'show' : 'hide'}`}>{this.props.detail}</p>
```